In [158]:
import pandas as pd
data=pd.read_csv("SelectedFeatures-10s-TOR-NonTOR.csv")
data

Source IP   Source Port   Destination IP   Destination Port  \
0            10.0.2.15         53913    216.58.208.46                 80   
1            10.0.2.15         53913    216.58.208.46                 80   
2            10.0.2.15         53913    216.58.208.46                 80   
3            10.0.2.15         53913    216.58.208.46                 80   
4            10.0.2.15         53913    216.58.208.46                 80   
...                ...           ...              ...                ...   
67829  131.202.240.183          7116  239.255.255.250               1900   
67830  131.202.240.183          7116  239.255.255.250               1900   
67831   131.202.240.87         11365       31.13.73.1                443   
67832     131.202.6.26         13000   131.202.240.87              64584   
67833   38.124.168.119            80   131.202.240.87              65089   

        Protocol   Flow Duration   Flow Bytes/s   Flow Packets/s  \
0              6             435            0.0      4597.701149   
1              6             259            0.0      7722.007722   
2              6             891            0.0      2244.668911   
3              6            1074            0.0      1862.197393   
4              6             315            0.0      6349.206349   
...          ...             ...            ...              ...   
67829         17               0            inf              inf   
67830         17               0            inf              inf   
67831          6               0            inf              inf   
67832          6               0            NaN              inf   
67833          6               0            NaN              inf   

        Flow IAT Mean   Flow IAT Std  ...   Bwd IAT Min  Active Mean  \
0               435.0            0.0  ...             0            0   
1               259.0            0.0  ...             0            0   
2               891.0            0.0  ...             0            0   
3              1074.0            0.0  ...             0            0   
4               315.0            0.0  ...             0            0   
...               ...            ...  ...           ...          ...   
67829             0.0            0.0  ...             0            0   
67830             0.0            0.0  ...             0            0   
67831             0.0            0.0  ...             0            0   
67832             0.0            0.0  ...             0            0   
67833             0.0            0.0  ...             0            0   

        Active Std   Active Max   Active Min  Idle Mean   Idle Std   Idle Max  \
0                0            0            0          0          0          0   
1                0            0            0          0          0          0   
2                0            0            0          0          0          0   
3                0            0            0          0          0          0   
4                0            0            0          0          0          0   
...            ...          ...          ...        ...        ...        ...   
67829            0            0            0          0          0          0   
67830            0            0            0          0          0          0   
67831            0            0            0          0          0          0   
67832            0            0            0          0          0          0   
67833            0            0            0          0          0          0   

        Idle Min   label  
0              0  nonTOR  
1              0  nonTOR  
2              0  nonTOR  
3              0  nonTOR  
4              0  nonTOR  
...          ...     ...  
67829          0  nonTOR  
67830          0  nonTOR  
67831          0  nonTOR  
67832          0  nonTOR  
67833          0  nonTOR  

[67834 rows x 29 columns]

In [159]:
data.columns

Index(['Source IP', ' Source Port', ' Destination IP', ' Destination Port',
       ' Protocol', ' Flow Duration', ' Flow Bytes/s', ' Flow Packets/s',
       ' Flow IAT Mean', ' Flow IAT Std', ' Flow IAT Max', ' Flow IAT Min',
       'Fwd IAT Mean', ' Fwd IAT Std', ' Fwd IAT Max', ' Fwd IAT Min',
       'Bwd IAT Mean', ' Bwd IAT Std', ' Bwd IAT Max', ' Bwd IAT Min',
       'Active Mean', ' Active Std', ' Active Max', ' Active Min', 'Idle Mean',
       ' Idle Std', ' Idle Max', ' Idle Min', 'label'],
      dtype='object')

In [160]:
data.groupby(['label']).count()

Source IP   Source Port   Destination IP   Destination Port  \
label                                                                 
TOR          8044          8044             8044               8044   
nonTOR      59790         59790            59790              59790   

         Protocol   Flow Duration   Flow Bytes/s   Flow Packets/s  \
label                                                               
TOR          8044            8044           8044             8044   
nonTOR      59790           59790          59788            59790   

         Flow IAT Mean   Flow IAT Std  ...   Bwd IAT Max   Bwd IAT Min  \
label                                  ...                               
TOR               8044           8044  ...          8044          8044   
nonTOR           59790          59790  ...         59790         59790   

        Active Mean   Active Std   Active Max   Active Min  Idle Mean  \
label                                                                   
TOR            8044         8044         8044         8044       8044   
nonTOR        59790        59790        59790        59790      59790   

         Idle Std   Idle Max   Idle Min  
label                                    
TOR          8044       8044       8044  
nonTOR      59790      59790      59790  

[2 rows x 28 columns]

In total 8044 connections are from TOR and 59790 from Other Browsers

In [161]:
data.groupby(['Source IP']).count()

Source Port   Destination IP   Destination Port   Protocol  \
Source IP                                                                     
1.11.38.90                 1                1                  1          1   
1.30.120.248               1                1                  1          1   
1.34.23.26                 1                1                  1          1   
1.9.106.104                1                1                  1          1   
10.0.2.15               6333             6333               6333       6333   
...                      ...              ...                ...        ...   
95.87.9.168                1                1                  1          1   
96.16.196.150              3                3                  3          3   
98.210.111.105             1                1                  1          1   
98.245.208.69              2                2                  2          2   
99.243.125.41              1                1                  1          1   

                 Flow Duration   Flow Bytes/s   Flow Packets/s  \
Source IP                                                        
1.11.38.90                   1              1                1   
1.30.120.248                 1              1                1   
1.34.23.26                   1              1                1   
1.9.106.104                  1              1                1   
10.0.2.15                 6333           6333             6333   
...                        ...            ...              ...   
95.87.9.168                  1              1                1   
96.16.196.150                3              3                3   
98.210.111.105               1              1                1   
98.245.208.69                2              2                2   
99.243.125.41                1              1                1   

                 Flow IAT Mean   Flow IAT Std   Flow IAT Max  ...  \
Source IP                                                     ...   
1.11.38.90                   1              1              1  ...   
1.30.120.248                 1              1              1  ...   
1.34.23.26                   1              1              1  ...   
1.9.106.104                  1              1              1  ...   
10.0.2.15                 6333           6333           6333  ...   
...                        ...            ...            ...  ...   
95.87.9.168                  1              1              1  ...   
96.16.196.150                3              3              3  ...   
98.210.111.105               1              1              1  ...   
98.245.208.69                2              2              2  ...   
99.243.125.41                1              1              1  ...   

                 Bwd IAT Min  Active Mean   Active Std   Active Max  \
Source IP                                                             
1.11.38.90                 1            1            1            1   
1.30.120.248               1            1            1            1   
1.34.23.26                 1            1            1            1   
1.9.106.104                1            1            1            1   
10.0.2.15               6333         6333         6333         6333   
...                      ...          ...          ...          ...   
95.87.9.168                1            1            1            1   
96.16.196.150              3            3            3            3   
98.210.111.105             1            1            1            1   
98.245.208.69              2            2            2            2   
99.243.125.41              1            1            1            1   

                 Active Min  Idle Mean   Idle Std   Idle Max   Idle Min  label  
Source IP                                                                       
1.11.38.90                1          1          1          1          1      1  
1.30.120.248              1          1          1          1     

787 Source IP

In [162]:
data.groupby(' Source Port').count()

Source IP   Destination IP   Destination Port   Protocol  \
 Source Port                                                             
21                    6                6                  6          6   
22                   15               15                 15         15   
53                    4                4                  4          4   
67                    2                2                  2          2   
68                    7                7                  7          7   
...                 ...              ...                ...        ...   
65527                 1                1                  1          1   
65528                 3                3                  3          3   
65530                 2                2                  2          2   
65532                 2                2                  2          2   
65534                 2                2                  2          2   

               Flow Duration   Flow Bytes/s   Flow Packets/s   Flow IAT Mean  \
 Source Port                                                                   
21                         6              6                6               6   
22                        15             15               15              15   
53                         4              4                4               4   
67                         2              2                2               2   
68                         7              7                7               7   
...                      ...            ...              ...             ...   
65527                      1              1                1               1   
65528                      3              3                3               3   
65530                      2              2                2               2   
65532                      2              2                2               2   
65534                      2              2                2               2   

               Flow IAT Std   Flow IAT Max  ...   Bwd IAT Min  Active Mean  \
 Source Port                                ...                              
21                        6              6  ...             6            6   
22                       15             15  ...            15           15   
53                        4              4  ...             4            4   
67                        2              2  ...             2            2   
68                        7              7  ...             7            7   
...                     ...            ...  ...           ...          ...   
65527                     1              1  ...             1            1   
65528                     3              3  ...             3            3   
65530                     2              2  ...             2            2   
65532                     2              2  ...             2            2   
65534                     2              2  ...             2            2   

               Active Std   Active Max   Active Min  Idle Mean   Idle Std  \
 Source Port                                                                
21                      6            6            6          6          6   
22                     15           15           15         15         15   
53                      4            4            4          4          4   
67                      2            2            2          2          2   
68                      7            7            7          7          7   
...                   ...          ...          ...        ...        ...   
65527                   1            1            1          1          1   
65528                   3            3            3          3          3   
65530                   2            2            2          2          2   
65532                   2            2            2          2          2   
65534                   2            2            2          2         

Traffic are from 17578 Ports

In [163]:
data.groupby([' Destination IP']).count()

Source IP   Source Port   Destination Port   Protocol  \
 Destination IP                                                          
1.28.240.176             1             1                  1          1   
10.0.2.15             2978          2978               2978       2978   
10.8.0.10              885           885                885        885   
10.8.0.14              234           234                234        234   
10.8.0.6               177           177                177        177   
...                    ...           ...                ...        ...   
99.252.169.119           1             1                  1          1   
99.29.0.18              21            21                 21         21   
99.4.177.15              1             1                  1          1   
99.67.53.71              3             3                  3          3   
99.94.171.150            1             1                  1          1   

                  Flow Duration   Flow Bytes/s   Flow Packets/s  \
 Destination IP                                                   
1.28.240.176                  1              1                1   
10.0.2.15                  2978           2978             2978   
10.8.0.10                   885            885              885   
10.8.0.14                   234            234              234   
10.8.0.6                    177            177              177   
...                         ...            ...              ...   
99.252.169.119                1              1                1   
99.29.0.18                   21             21               21   
99.4.177.15                   1              1                1   
99.67.53.71                   3              3                3   
99.94.171.150                 1              1                1   

                  Flow IAT Mean   Flow IAT Std   Flow IAT Max  ...  \
 Destination IP                                                ...   
1.28.240.176                  1              1              1  ...   
10.0.2.15                  2978           2978           2978  ...   
10.8.0.10                   885            885            885  ...   
10.8.0.14                   234            234            234  ...   
10.8.0.6                    177            177            177  ...   
...                         ...            ...            ...  ...   
99.252.169.119                1              1              1  ...   
99.29.0.18                   21             21             21  ...   
99.4.177.15                   1              1              1  ...   
99.67.53.71                   3              3              3  ...   
99.94.171.150                 1              1              1  ...   

                  Bwd IAT Min  Active Mean   Active Std   Active Max  \
 Destination IP                                                        
1.28.240.176                1            1            1            1   
10.0.2.15                2978         2978         2978         2978   
10.8.0.10                 885          885          885          885   
10.8.0.14                 234          234          234          234   
10.8.0.6                  177          177          177          177   
...                       ...          ...          ...          ...   
99.252.169.119              1            1            1            1   
99.29.0.18                 21           21           21           21   
99.4.177.15                 1            1            1            1   
99.67.53.71                 3            3            3            3   
99.94.171.150               1            1            1            1   

                  Active Min  Idle Mean   Idle Std   Idle Max   Idle Min  \
 Destination IP                                                            
1.28.240.176               1          1          1          1          1   
10.0.2.15               2978       2978       2978       2978       2978   
10.8.0.10                885  

1736 Destination IP

In [164]:
data.groupby([' Destination Port']).count()

Source IP   Source Port   Destination IP   Protocol  \
 Destination Port                                                        
21                       103           103              103        103   
22                       170           170              170        170   
23                         4             4                4          4   
53                     12267         12267            12267      12267   
67                         7             7                7          7   
...                      ...           ...              ...        ...   
65358                      1             1                1          1   
65364                      1             1                1          1   
65381                      7             7                7          7   
65397                      1             1                1          1   
65514                     21            21               21         21   

                    Flow Duration   Flow Bytes/s   Flow Packets/s  \
 Destination Port                                                   
21                            103            103              103   
22                            170            170              170   
23                              4              4                4   
53                          12267          12267            12267   
67                              7              7                7   
...                           ...            ...              ...   
65358                           1              1                1   
65364                           1              1                1   
65381                           7              7                7   
65397                           1              1                1   
65514                          21             21               21   

                    Flow IAT Mean   Flow IAT Std   Flow IAT Max  ...  \
 Destination Port                                                ...   
21                            103            103            103  ...   
22                            170            170            170  ...   
23                              4              4              4  ...   
53                          12267          12267          12267  ...   
67                              7              7              7  ...   
...                           ...            ...            ...  ...   
65358                           1              1              1  ...   
65364                           1              1              1  ...   
65381                           7              7              7  ...   
65397                           1              1              1  ...   
65514                          21             21             21  ...   

                    Bwd IAT Min  Active Mean   Active Std   Active Max  \
 Destination Port                                                        
21                          103          103          103          103   
22                          170          170          170          170   
23                            4            4            4            4   
53                        12267        12267        12267        12267   
67                            7            7            7            7   
...                         ...          ...          ...          ...   
65358                         1            1            1            1   
65364                         1            1            1            1   
65381                         7            7            7            7   
65397                         1            1            1            1   
65514                        21           21           21           21   

                    Active Min  Idle Mean   Idle Std   Idle Max   Idle Min  \
 Destination Port                                                            
21                         103        103        103        103        103   
22                      

2215  Destination Port

In [165]:
from sklearn import preprocessing
le = preprocessing.LabelEncoder()
data[' Source Port'] = le.fit_transform(data[' Source Port'])

In [166]:
dp = preprocessing.LabelEncoder()
data[' Destination Port'] = dp.fit_transform(data[' Destination Port'])

In [167]:
tp = preprocessing.LabelEncoder()
data['label'] = tp.fit_transform(data['label'])

In [168]:
data.head()

Source IP   Source Port  Destination IP   Destination Port   Protocol  \
0  10.0.2.15         11873   216.58.208.46                  6          6   
1  10.0.2.15         11873   216.58.208.46                  6          6   
2  10.0.2.15         11873   216.58.208.46                  6          6   
3  10.0.2.15         11873   216.58.208.46                  6          6   
4  10.0.2.15         11873   216.58.208.46                  6          6   

    Flow Duration   Flow Bytes/s   Flow Packets/s   Flow IAT Mean  \
0             435            0.0      4597.701149           435.0   
1             259            0.0      7722.007722           259.0   
2             891            0.0      2244.668911           891.0   
3            1074            0.0      1862.197393          1074.0   
4             315            0.0      6349.206349           315.0   

    Flow IAT Std  ...   Bwd IAT Min  Active Mean   Active Std   Active Max  \
0            0.0  ...             0            0            0            0   
1            0.0  ...             0            0            0            0   
2            0.0  ...             0            0            0            0   
3            0.0  ...             0            0            0            0   
4            0.0  ...             0            0            0            0   

    Active Min  Idle Mean   Idle Std   Idle Max   Idle Min  label  
0            0          0          0          0          0      1  
1            0          0          0          0          0      1  
2            0          0          0          0          0      1  
3            0          0          0          0          0      1  
4            0          0          0          0          0      1  

[5 rows x 29 columns]

In [169]:
data_source=data.drop(columns=['Source IP',' Destination IP', ' Source Port',' Destination Port'])

In [170]:
data_source.columns

Index([' Protocol', ' Flow Duration', ' Flow Bytes/s', ' Flow Packets/s',
       ' Flow IAT Mean', ' Flow IAT Std', ' Flow IAT Max', ' Flow IAT Min',
       'Fwd IAT Mean', ' Fwd IAT Std', ' Fwd IAT Max', ' Fwd IAT Min',
       'Bwd IAT Mean', ' Bwd IAT Std', ' Bwd IAT Max', ' Bwd IAT Min',
       'Active Mean', ' Active Std', ' Active Max', ' Active Min', 'Idle Mean',
       ' Idle Std', ' Idle Max', ' Idle Min', 'label'],
      dtype='object')

In [171]:
flags_ohe = data_source
categorical_columns = [' Protocol']
for col in categorical_columns:
    col_ohe = pd.get_dummies(data_source[col], prefix=col)
    flags_ohe = pd.concat((flags_ohe, col_ohe), axis=1).drop(col, axis=1)

In [172]:
import numpy as np
flags_ohe=flags_ohe.fillna(0)
from sklearn import preprocessing
  
# label_encoder object knows how to understand word labels.
label_encoder = preprocessing.LabelEncoder()
  
# Encode labels in column 'species'.
flags_ohe['label']= label_encoder.fit_transform(flags_ohe['label'])
  
flags_ohe['label'].unique()

array([1, 0])

In [173]:
for i in flags_ohe.columns:
  flags_ohe=flags_ohe[(flags_ohe[i]!='inf')]

In [174]:
flags_ohe.drop(flags_ohe.tail(6).index,
        inplace = True)

In [175]:
from sklearn.model_selection import train_test_split

#train, test = train_test_split(flags_ohe, test_size = 0.1)
df = flags_ohe.fillna(0)
train_x, test_x, train_y, test_y = train_test_split(df[df.columns.drop(['label'])], df['label'],test_size=0.1)

In [176]:

# example of a standardization
from numpy import asarray
from sklearn.preprocessing import StandardScaler
train_x = train_x.astype(np.float32)
col = train_x.columns
scaler = StandardScaler()
train_x = scaler.fit_transform(train_x)
test_x = scaler.transform(test_x)

In [177]:
from sklearn.linear_model import SGDClassifier, SGDRegressor,LogisticRegression
from sklearn.ensemble import RandomForestClassifier, GradientBoostingClassifier
from sklearn.naive_bayes import MultinomialNB
from sklearn.metrics import accuracy_score, classification_report, confusion_matrix
import random
import itertools

In [178]:

train_x = pd.DataFrame(train_x, columns = col)
test_x = pd.DataFrame(test_x, columns = col)


Logistic Regression

In [179]:
basicmodel = LogisticRegression()
basicmodel = basicmodel.fit(train_x, train_y)
preds1 = basicmodel.predict(test_x)
acc1=accuracy_score(test_y, preds1)
print("Accuracy is {}".format(acc1*100))
print(classification_report(test_y, preds1))
print(confusion_matrix(test_y, preds1))

Accuracy is 94.0439333628188
              precision    recall  f1-score   support

           0       0.79      0.70      0.74       836
           1       0.96      0.97      0.97      5947

    accuracy                           0.94      6783
   macro avg       0.88      0.84      0.85      6783
weighted avg       0.94      0.94      0.94      6783

[[ 583  253]
 [ 151 5796]]


/usr/local/lib/python3.9/dist-packages/sklearn/linear_model/_logistic.py:458: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


Random Forest

In [181]:
basicmodel = RandomForestClassifier()
basicmodel = basicmodel.fit(train_x, train_y)
preds1 = basicmodel.predict(test_x)
acc1=accuracy_score(test_y, preds1)
print("Accuracy is {}".format(acc1*100))
print(classification_report(test_y, preds1))
print(confusion_matrix(test_y, preds1))

Accuracy is 88.19106590004424
              precision    recall  f1-score   support

           0       0.89      0.05      0.09       836
           1       0.88      1.00      0.94      5947

    accuracy                           0.88      6783
   macro avg       0.89      0.52      0.51      6783
weighted avg       0.88      0.88      0.83      6783

[[  40  796]
 [   5 5942]]


Gradient Boosting Machines

In [182]:
  basicmodel = GradientBoostingClassifier()
basicmodel = basicmodel.fit(train_x, train_y)
preds1 = basicmodel.predict(test_x)
acc1=accuracy_score(test_y, preds1)
print("Accuracy is {}".format(acc1*100))
print(classification_report(test_y, preds1))
print(confusion_matrix(test_y, preds1))

Accuracy is 86.83473389355743
              precision    recall  f1-score   support

           0       0.44      0.27      0.34       836
           1       0.90      0.95      0.93      5947

    accuracy                           0.87      6783
   macro avg       0.67      0.61      0.63      6783
weighted avg       0.85      0.87      0.85      6783

[[ 229  607]
 [ 286 5661]]


Stochastic Gradient Descent Classifier

In [183]:
basicmodel= SGDClassifier(loss='modified_huber', max_iter=5, random_state=0, shuffle=True)
basicmodel = basicmodel.fit(train_x, train_y)
preds1 = basicmodel.predict(test_x)
acc1=accuracy_score(test_y, preds1)
print("Accuracy is {}".format(acc1*100))
print(classification_report(test_y, preds1))
print(confusion_matrix(test_y, preds1))

Accuracy is 87.55712811440365
              precision    recall  f1-score   support

           0       0.28      0.01      0.01       836
           1       0.88      1.00      0.93      5947

    accuracy                           0.88      6783
   macro avg       0.58      0.50      0.47      6783
weighted avg       0.80      0.88      0.82      6783

[[   5  831]
 [  13 5934]]


/usr/local/lib/python3.9/dist-packages/sklearn/linear_model/_stochastic_gradient.py:702: ConvergenceWarning: Maximum number of iteration reached before convergence. Consider increasing max_iter to improve the fit.
  warnings.warn(


Support Vector Machine

In [184]:
from sklearn import svm
basicmodel = svm.SVC(kernel='sigmoid')
basicmodel.fit(train_x, train_y)
preds1 = basicmodel.predict(test_x)
acc1=accuracy_score(test_y, preds1)
print("Accuracy is {}".format(acc1*100))
print(classification_report(test_y, preds1))
print(confusion_matrix(test_y, preds1))

Accuracy is 89.65059708093763
              precision    recall  f1-score   support

           0       0.58      0.61      0.59       836
           1       0.95      0.94      0.94      5947

    accuracy                           0.90      6783
   macro avg       0.76      0.77      0.77      6783
weighted avg       0.90      0.90      0.90      6783

[[ 513  323]
 [ 379 5568]]


XG Boost

In [187]:
import xgboost as xgb
xgb_model = xgb.XGBClassifier(objective="binary:logistic", random_state=42)
xgb_model.fit(train_x, train_y)

y_pred = xgb_model.predict(test_x)
acc1=accuracy_score(test_y, y_pred)
print("Accuracy is {}".format(acc1*100))
print(classification_report(test_y, y_pred))
print(confusion_matrix(test_y, y_pred))

Accuracy is 88.08786672563762
              precision    recall  f1-score   support

           0       0.53      0.27      0.36       836
           1       0.90      0.97      0.93      5947

    accuracy                           0.88      6783
   macro avg       0.72      0.62      0.65      6783
weighted avg       0.86      0.88      0.86      6783

[[ 229  607]
 [ 201 5746]]


AdaBoost

In [186]:
from sklearn.ensemble import AdaBoostClassifier
abc = AdaBoostClassifier(n_estimators=50,
                         learning_rate=1)
model=abc.fit(train_x, train_y)

y_pred = model.predict(test_x)
acc1=accuracy_score(test_y, y_pred)
print("Accuracy is {}".format(acc1*100))
print(classification_report(test_y, y_pred))
print(confusion_matrix(test_y, y_pred))

Accuracy is 90.2108211705735
              precision    recall  f1-score   support

           0       0.74      0.31      0.44       836
           1       0.91      0.98      0.95      5947

    accuracy                           0.90      6783
   macro avg       0.83      0.65      0.69      6783
weighted avg       0.89      0.90      0.88      6783

[[ 262  574]
 [  90 5857]]


AdaBoost

In [188]:
from sklearn.ensemble import AdaBoostClassifier
abc = AdaBoostClassifier(n_estimators=50,
                         learning_rate=1)
model=abc.fit(train_x, train_y)
y_pred = model.predict(test_x)
acc1=accuracy_score(test_y, y_pred)
print("Accuracy is {}".format(acc1*100))
print(classification_report(test_y, y_pred))
print(confusion_matrix(test_y, y_pred))

Accuracy is 90.2108211705735
              precision    recall  f1-score   support

           0       0.74      0.31      0.44       836
           1       0.91      0.98      0.95      5947

    accuracy                           0.90      6783
   macro avg       0.83      0.65      0.69      6783
weighted avg       0.89      0.90      0.88      6783

[[ 262  574]
 [  90 5857]]


KNN

In [190]:
from sklearn.neighbors import KNeighborsClassifier
knn = KNeighborsClassifier(n_neighbors=3)
knn.fit(train_x, train_y)
y_pred = knn.predict(test_x)
acc1=accuracy_score(test_y, y_pred)
print("Accuracy is {}".format(acc1*100))
print(classification_report(test_y, y_pred))
print(confusion_matrix(test_y, y_pred))

Accuracy is 95.53295002211411
              precision    recall  f1-score   support

           0       0.84      0.78      0.81       836
           1       0.97      0.98      0.97      5947

    accuracy                           0.96      6783
   macro avg       0.91      0.88      0.89      6783
weighted avg       0.95      0.96      0.95      6783

[[ 653  183]
 [ 120 5827]]
